kornia/augmentation/_2d/intensity/channel_dropout.py

```diff
@@ -0,0 +1,89 @@
+        same_on_batch: Apply the same transformation across the batch. Defaults to False.
+        p: Probability of applying the transformation. Defaults to 0.5.
+        keepdim: Whether to keep the output shape the same as input ``True`` or broadcast it
+            to the batch form ``False``. Defaults to False.
+
+    Shape:
+        - Input: :math:`(3, H, W)` or :math:`(B, 3, H, W)`
+        - Output: :math:`(3, H, W)` or :math:`(B, 3, H, W)`
+
+    .. note::
+        If `num_drop_channels` is set to 1, it means that for each image in the batch,
+            we will randomly choose one channel to drop.
+        If `num_drop_channels` is set to 2, it means that for each image in the batch,
+            we will randomly choose two channels to drop.
+        If num_drop_channels is set to 3, it means that for each image in the batch,
+            we will randomly choose three channels to drop.
+
+    Examples:
+        >>> rng = torch.manual_seed(1)
+        >>> img = torch.ones(1, 3, 3, 3)
+        >>> aug = RandomChannelDropout(num_drop_channels=1, fill_value=0.0, p=1.0)
+        >>> aug(img)
+        tensor([[[[1., 1., 1.],
+                  [1., 1., 1.],
+                  [1., 1., 1.]],
+        <BLANKLINE>
+                 [[0., 0., 0.],
+                  [0., 0., 0.],
+                  [0., 0., 0.]],
+        <BLANKLINE>
+                 [[1., 1., 1.],
+                  [1., 1., 1.],
+                  [1., 1., 1.]]]])
+
+    To apply the exact augmenation again, you may take the advantage of the previous parameter state:
+    >>> input = torch.rand(1, 3, 32, 32)
+    >>> aug = RandomChannelDropout(num_drop_channels=1, fill_value=0.0, p=1.0)
+    >>> (aug(input) == aug(input, params=aug._params)).all()
+    tensor(True)
+    """
+
+    def __init__(
+        self,
+        num_drop_channels: int = 1,
+        fill_value: float = 0.0,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim)
+
+        KORNIA_CHECK(
+            0.0 <= fill_value <= 1.0, f"Invalid fill_value. Should be a float bewteen 0 and 1. Got: {fill_value}"
+        )
+        self.fill_value = fill_value
+
+        KORNIA_CHECK(
+            1 <= num_drop_channels <= 3,
+            f"Invalid num_drop_channels. Should be a int bewteen 1 and 3. Got: {num_drop_channels}",
+        )
+        self.num_drop_channels = num_drop_channels
+        # Generator of random parameters.
+        self._param_generator = ChannelDropoutGenerator(self.num_drop_channels)
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        KORNIA_CHECK_SHAPE(input, ["*", "3", "H", "W"])
+
+        out = input.clone()
+        out[params["dropout_mask"].to(device=out.device)] = self.fill_value
```

Comment:
@vgilabert94 have you benchmark tried directly just to keep the indices (instead of allocating a mask) and just torch fill_ ? @johnnv1 this could be a good use case to start putting in practice the ops benchmark api we discussed

Reply:
If you can provide me a snippet code or some example of how i can do it.

Reply:
```python
fill_value = 0
inpt = torch.rand(1,4, 5, 3)
channels_to_fill = [0, 2]
out = inpt.clone()
return out[..., channels_to_fill, :, : ] = fill_value
```

Reply:
about the performance, i remember seeing something in an old PR, but I didn't find it 😞 

you can benchmark your augmentation with https://pytorch.org/tutorials/recipes/recipes/benchmark.html -- here: https://github.com/kornia/kornia-benchmark  have some examples  

will look something like
```python
from kornia.augmentation import RandomChannelDropout
import torch
from torch.utils import benchmark


aug = RandomChannelDropout()
inpt = torch.rand(5, 3, 512, 512, device=torch.device('cuda'))

result = benchmark.Timer(
    stmt="aug(inpt)",
    setup="from __main__ import aug",
    globals={
        "image": inpt,
    },
    description='RandomChannelDropout',
).timeit(1000)

print(result)
```

if you have two cases/results to check you can do  

```python
from kornia.augmentation import RandomChannelDropout
import torch
from torch.utils import benchmark

aug_A = RandomChannelDropout_A()
aug_B = RandomChannelDropout_B()
inpt = torch.rand(5, 3, 512, 512, device=torch.device('cuda'))

results = [
  benchmark.Timer(
      stmt="aug_A(inpt)",
      setup="from __main__ import aug_A",
      globals={
          "image": inpt,
      },
      description='RandomChannelDropout_A',
  ).timeit(1000),
  benchmark.Timer(
      stmt="aug_B(inpt)",
      setup="from __main__ import aug_B",
      globals={
          "image": inpt,
      },
      description='RandomChannelDropout_B',
  ).timeit(1000),
]

compare = benchmark.Compare(results)
compare.print()
```